core/src/main/scala/org/typelevel/paiges/Chunk.scala

```diff
@@ -40,15 +35,46 @@ private[paiges] object Chunk {
       }
     }
 
+    class LineCombiner {
+      var line: StringBuilder = new StringBuilder
+      def nonEmpty = line.nonEmpty
+      def finalLine() = {
+        val res = line.toString
+        line = new StringBuilder
+        LineCombiner.trim(res)
+      }
+      def addItem(item: ChunkStream.Item): Option[String] =
+        if (!trim) {
+          Some(item.stringChunk)
+        } else if (item.isBreak) {
+          val v = LineCombiner.trim(line.toString)
+          line = new StringBuilder(lineToStr(item.position))
+          Some(v)
+        } else {
+          line.append(item.str)
+          None
+        }
+    }
+
+    object LineCombiner {
+      def trim(s: String) = {
+        var ind = s.length
+        while (ind >= 1 && s.charAt(ind - 1) == ' ') ind = ind - 1
+        s.substring(0, ind)
+      }
+    }
+
     class ChunkIterator(var current: ChunkStream) extends Iterator[String] {
-      def hasNext: Boolean = (current != ChunkStream.Empty)
-      def next: String = {
-        val item = current.asInstanceOf[ChunkStream.Item]
-        val res = item.stringChunk
-        current = item.step
-        res
+      private val lineCombiner = new LineCombiner
```

Comment:
can't we skip this code path entirely if there is no trim? If we have two subclasses of Iterator, one for trim, one without, my guess is that we will not slow down much because we can at the outer-most scope check to see if we need to make a trimming iterator.

Reply:
I was hoping that the JIT would do that for us, but sure, can add that.

Reply:
Done.